### ISSUE TITLE
Add exercises/transformer_ts.ipynb + solution: Keras self-attention on FordA time series (DL2026 Phase 5)

### ISSUE THREAD

**Issue by yoavram** [yoavram/nanochat#1]
## What

Add `exercises/transformer_ts.ipynb` and `solutions/transformer_ts.ipynb` — a Keras
self-attention classifier on the FordA time-series dataset, as homework for Day 5.

This was Phase 5 of the DL2026 restructuring of [yoavram/DataSciPy](https://github.com/yoavram/DataSciPy),
postponed rather than dropped. Every other phase has landed; this is the only outstanding
item, and it is the one piece of that plan whose deliverable belongs in *this* repository.

## Why here, and why after `text-transformer.ipynb`

Days 5–6 live in this repo, and this exercise is meant to follow `text-transformer.ipynb`
in the Day 5 sequence. The placement is the point: **same architecture, different data,
and Keras instead of JAX.** Students meet self-attention on text in JAX, then immediately
apply it to a time series in Keras, which separates the idea from both the framework and
the modality it was introduced with.

## Source material

`sessions/transformer_ts.ipynb` exists on `DataSciPy@master` and is the starting point:

```bash
git show origin/master:sessions/transformer_ts.ipynb > transformer_ts.ipynb
```

It is 27 cells / 13 code cells, imports only `jax`, `keras`, `matplotlib`, `numpy` and
`pickle`, and is structured as **Data → Self-attention → Transformer classifier →
References → Colophon**. It reads `../data/FordA_TRAIN.tsv` and `../data/FordA_TEST.tsv`
(both small and committed on the DataSciPy side) and checkpoints to
`../data/ford_transformer.keras` plus a history pickle.

Note it was deliberately **not** committed to the `DL2026` branch — the branch assembly
copied it out to scratch precisely so it could become an exercise here rather than
staying a session there.

## The actual work

This is authoring, not a file move. Converting a walkthrough into an assignment means:

1. **Strip the completed self-attention block and the classifier construction into TODOs.**
   Those two are the exercise; everything else is scaffolding.
2. **Keep data loading and evaluation intact.** A student should be able to run the
   notebook top to bottom, hit the TODOs, and have a working evaluation waiting for them
   once they fill them in.
3. **Write `solutions/transformer_ts.ipynb` separately**, matching the convention that
   every exercise in these courses has a paired solution.
4. Neither `exercises/` nor `solutions/` exists in this repo yet — all notebooks are in the
   root — so both directories need creating, and `index.ipynb` needs the
   `[assignment](...) | [solution](...)` link pair under Day 5.

## Conventions to match

- Opening cell with the logo, title, `## Yoav Ram`, and an "In this session we will
  understand:" list; `# References` and `# Colophon` (CC BY-NC-SA) closing cells.
- Markdown narrative between every code cell — no runs of bare code.
- Fixed `SEED`; print the Keras version and backend in the first code cell.
- Use *validation* terminology rather than *test* (`X_validation`, `val_accuracy`). The
  source notebook predates that sweep on the DataSciPy side, so this needs applying.
- Keras 3 on the JAX backend. Do not introduce `torch`, `tensorflow` or
  `tensorflow_datasets`.
- Commit the notebooks **with** their outputs; the figures are part of the material.
- Checkpoints are gitignored, so pair every long training cell with a working
  `load_model` path.

## One thing worth deciding first

The FordA data files live in `DataSciPy/data/`. This repo will need its own copy or its own
download step — they are small enough to commit, but that is a call to make rather than
assume.
